src/ol/renderer/Layer.js

```diff
@@ -196,11 +195,7 @@ class LayerRenderer extends Observable {
    */
   renderIfReadyAndVisible() {
     const layer = this.getLayer();
-    if (
-      layer &&
-      layer.getVisible() &&
-      layer.getSourceState() == SourceState.READY
-    ) {
+    if (layer && layer.getVisible() && 'ready') {
```

Comment:
```suggestion
    if (layer && layer.getVisible() && layer.getSourceState() == 'ready') {
```
The correct comparison was lost here.

Reply:
Good find, @MoonE, thanks! I created #13835 to fix this, and also added a regression test.